ingestion-beam/src/main/java/com/mozilla/telemetry/IpPrivacyDecoder.java

```diff
@@ -0,0 +1,95 @@
+import com.mozilla.telemetry.decoder.Deduplicate;
+import com.mozilla.telemetry.decoder.GeoCityLookup;
+import com.mozilla.telemetry.decoder.ParseProxy;
+import com.mozilla.telemetry.decoder.ParseUri;
+import com.mozilla.telemetry.decoder.ipprivacy.HashClientInfo;
+import com.mozilla.telemetry.decoder.ipprivacy.IpParsePayload;
+import com.mozilla.telemetry.decoder.ipprivacy.ParseIp;
+import com.mozilla.telemetry.decoder.ipprivacy.RemoveAttributes;
+import com.mozilla.telemetry.ingestion.core.Constant.Attribute;
+import com.mozilla.telemetry.transforms.DecompressPayload;
+import com.mozilla.telemetry.transforms.DeduplicateByDocumentId;
+import com.mozilla.telemetry.transforms.LimitPayloadSize;
+import com.mozilla.telemetry.transforms.NormalizeAttributes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.Filter;
+import org.apache.beam.sdk.transforms.Flatten;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionList;
+
+/**
+ * Job for generating the IP privacy dataset that is based on the decoder job with
+ * unneeded steps taken out.
+ */
+public class IpPrivacyDecoder extends Sink {
+
+  /**
+   * Execute an Apache Beam pipeline.
+   *
+   * @param args command line arguments
+   */
+  public static void main(String[] args) {
+    run(args);
+  }
+
+  /**
+   * Execute an Apache Beam pipeline and return the {@code PipelineResult}.
+   *
+   * @param args command line arguments
+   */
+  public static PipelineResult run(String[] args) {
+    registerOptions(); // Defined in Sink.java
+    final DecoderOptions.Parsed options = DecoderOptions.parseDecoderOptions(
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(DecoderOptions.class));
+    return run(options);
+  }
+
+  /**
+   * Execute an Apache Beam pipeline and return the {@code PipelineResult}.
+   */
+  public static PipelineResult run(DecoderOptions.Parsed options) {
+    final Pipeline pipeline = Pipeline.create(options);
+    final List<PCollection<PubsubMessage>> errorCollections = new ArrayList<>();
+
+    // We wrap pipeline in Optional for more convenience in chaining together transforms.
+    Optional.of(pipeline) //
+        .map(p -> p //
+            .apply(options.getInputType().read(options)) //
+            .apply(ParseUri.of()).failuresTo(errorCollections) //
+            .apply("RestrictToMainPings", Filter.by((message) -> {
+              String doctype = message.getAttribute(Attribute.DOCUMENT_TYPE);
+              return doctype != null && doctype.equals("main");
+            })).apply(ParseProxy.of()) //
+            .apply(ParseIp.of()) //
+            .apply(GeoCityLookup.of(options.getGeoCityDatabase(), options.getGeoCityFilter())) //
```

Comment:
This is pointing out some refactoring we should probably do in the future. GeoCityLookup implicitly throws out the IP, and that should potentially be made explicit with its own transform. Definitely not in scope for this PR.